Installer.php

```diff
@@ -191,49 +190,26 @@ protected function loadExtensions()
         if (!is_file($file)) {
             return [];
         }
-        // invalidate opcache of extensions.php if exists
-        if (function_exists('opcache_invalidate')) {
-            @opcache_invalidate($file, true);
-        }
-        $extensions = require($file);
-
-        $vendorDir = str_replace('\\', '/', $this->vendorDir);
-        $n = strlen($vendorDir);
-
-        foreach ($extensions as &$extension) {
-            if (isset($extension['alias'])) {
-                foreach ($extension['alias'] as $alias => $path) {
-                    $path = str_replace('\\', '/', $path);
-                    if (strpos($path . '/', $vendorDir . '/') === 0) {
-                        $extension['alias'][$alias] = '<vendor-dir>' . substr($path, $n);
-                    }
-                }
-            }
-        }
 
-        return $extensions;
+        return (array) require($file);
     }
 
     protected function saveExtensions(array $extensions)
     {
         $file = $this->vendorDir . '/' . static::EXTENSION_FILE;
-        if (!file_exists(dirname($file))) {
-            mkdir(dirname($file), 0777, true);
-        }
-        $array = str_replace("'<vendor-dir>", '$vendorDir . \'', var_export($extensions, true));
-        file_put_contents($file, "<?php\n\n\$vendorDir = dirname(__DIR__);\n\nreturn $array;\n");
+        $this->filesystem->ensureDirectoryExists(dirname($file));
+        $array = str_replace("'$this->vendorDir/", '$vendorDir . \'/', var_export($extensions, true));
         // invalidate opcache of extensions.php if exists
         if (function_exists('opcache_invalidate')) {
-            @opcache_invalidate($file, true);
+            opcache_invalidate($file, true);
```

Comment:
That would make Installer fail if extensions.php does not exist.

Reply:
@samdark `opcache_invalidate()` don't generate error